### ISSUE TITLE
Can I use IndexedDbVFS in my application?

### ISSUE THREAD

**Issue by rhashimoto** [rhashimoto/wa-sqlite#11]
You probably shouldn't use IndexedDbVFS as-is in your application.

The example classes primarily exist for testing and to provide easy-to-understand..._examples_. They are not necessarily appropriate for use in a production application. In particular, the example IndexedDB VFS is _not_ well-suited for that.

The weakness in IndexedDbVFS is its locking, the feature that ensures that simultaneous database connections work safely and consistently. There's another question about [implementing locking](https://github.com/rhashimoto/wa-sqlite/issues/5), but basically IndexedDbVFS chose the way that demonstrates the concept while keeping the class short and simple.

If your application doesn't require locking, either because you never have multiple database connections or because your application prevent unsafe simultaneous access, then the locking in IndexedDbVFS is an unnecessary cost that could drag down performance. If your application does require locking, then the locking in IndexedDbVFS is almost certainly not sophisticated enough as it is prone to starvation and orphaned locks.

The bottom line is that if you want to use SQLite with IndexedDB storage, you can start with IndexedDbVFS but you should modify it for the needs of your application. Does your application require locking at all? Does your application connect to the same database from multiple contexts? From multiple tabs? What browsers do you need to support? The answers to these questions should direct your approach.
